ledger/test-common/src/main/scala/com.digitalasset.platform.testing/WaitForCompletionsObserver.scala

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.testing
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import com.digitalasset.ledger.api.v1.command_completion_service.CompletionStreamResponse
+import io.grpc.Context
+import io.grpc.stub.StreamObserver
+
+import scala.concurrent.{Future, Promise}
+
+object WaitForCompletionsObserver {
```

Comment:
This seems to be basically the same as `SizeBoundObserver`, with the only real difference being that this one fails when `onCompleted` is called, and `SizeBoundObserver` succeeds. Can you make that one do what you want instead?

Reply:
Not really, this is specialized to count the number of completions within each response, not to count each response as one.